pkg/login/social/azuread_oauth.go

```diff
@@ -56,7 +72,12 @@ func (s *SocialAzureAD) UserInfo(_ *http.Client, token *oauth2.Token) (*BasicUse
 	role := extractRole(claims, s.autoAssignOrgRole)
 	logger.Debug("AzureAD OAuth: extracted role", "email", email, "role", role)
 
-	groups := extractGroups(claims)
+	groups, err := extractGroups(client, claims, token)
+	if err != nil {
+		logger.Error("AzureAD OAuth: failed to extract groups", "error", err)
+		return nil, err
```

Comment:
No need to log the error here, pass it up to the caller and let the caller handle it.

```suggestion
		return nil, fmt.Errorf("failed to extract groups: %w", err)
```